### ISSUE TITLE
A unit can only have 1 working stockpile weapon

### ISSUE THREAD

**Issue by sprunk** [beyond-all-reason/spring#1817]
* one of the stockpile weapons will be assigned as "the" stockpile weapon on a unit.
* Lua interfaces apply to this weapon and you can't specify a different one.
* native stockpile command will only apply to ammo for that one weapon. This is probably fine if you can use Lua, but the command could have a weapon index parameter.
* the other weapons are still technically stockpile weapons so can't fire due to being stuck at 0 ammo.

**Comment by ryanbennitt** [comment]
Yeah you'd have to have multiple stockpile buttons if multiple weapons needed different stockpiles. Interesting other weapons don't work, they don't even have the stockpile=true flag.
